### PR TITLE
Add unit tests for modal.js

### PR DESCRIPTION
**Pull request by aayushmau5**
#1507 

This commit:
- Adds unit tests for the modal.js component
- Adds aria labels in component for accessibility purposes.

### CODE REVIEW

File: app/javascript/components/project-submissions/components/__tests__/modal.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    const { getByLabelText } = render(
+      <Modal 
+        handleClose={() => {}} 
+        show={false}
+      >
+        {" "}
+      </Modal>
+    );
+
+    expect(getByLabelText("modal--hidden")).toHaveClass("react-modal--hidden");
```
Comment (leosoaivan) [inline app/javascript/components/project-submissions/components/__tests__/modal.test.js:16]: Test should not rely solely on an assertion on a class name, which is opaque and unnecessary to a user. Add an assertion that tests for the presence/absence of children, per line 14 of the component. You may want to pass in a simple `div` with some text to assert against.

File: app/javascript/components/project-submissions/components/__tests__/modal.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const { getByLabelText } = render(
+      <Modal 
+        handleClose={() => {}} 
+        show={true}
+      >
+        {" "}
+      </Modal>
+    );
+
+    expect(getByLabelText("modal--shown")).toHaveClass("react-modal");
```
Comment (leosoaivan) [inline app/javascript/components/project-submissions/components/__tests__/modal.test.js:29]: Same here: Add an assertion that tests for the presence/absence of children, per line 14 of the component.

**Review by leosoaivan — CHANGES_REQUESTED** [review]
Excellent start.

**Comment by aayushmau5** [comment]
@leosoaivan Please check now.

**Review by leosoaivan — APPROVED** [review]

**Pull request merged**